Improve chrashlog

miguelinux opened the pull request:
Hi @chengangc ,

This is my start of the proposal, it is no finished, I count finish today, i will finish it tomorrow.

I'm send you this PR to you so we can review this proposal and then when you aproval send to the mailing list or directly to the upstream repo.

miguelinux committed:
crashlog: do not alter system behavior with watchdog

Instead of installing to systemd config directory, install to
/usr/share/acrn/crashlog. Using this way we not alter the system if it
is not used as a debug platform.

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

miguelinux committed:
crashlog: re-write usercrash-wrapper

Using an O(n^2) function for look up the values from the arguments
simplify the complexity code of the userchrash-wrapper.

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

miguelinux committed:
crashlog: introducing crashlogctl

This patch introduce the command `crashlogctl`,
it is used to enable/disable the ACRN crashlog services.

Also a systemd-tmpfile config file is introduced to create the
/var/log/crashlog directory, and a sysctl config file where
the usercrash-wrapper will be configured at boot time.

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

chengangc commented:
Thanks a lot for your help, Miguel.

So in your following patch/es, there will be somewhere to handle the watchdog.conf file?

miguelinux committed:
crashlog: deprecate acrnprobe_prepare and update Makefile

The `crashlogctl` command is replacing acrnprobe_prepare script.
So, let's remove from the project, and update the Makefile with
the new content.

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

miguelinux commented:
Hi @chengangc 

Just finished the patch set, please test it and let me know any concerns.

I rebased to the current ACRN master tree.

sorry for this delay

chengangc commented on tools/acrn-crashlog/Makefile:
```diff
@@ -136,15 +138,18 @@ uninstall:
 	@if [ -e "$(DESTDIR)/usr/share/acrn/crashlog/40-watchdog.conf" ];then \
 		$(RM) $(DESTDIR)/usr/share/acrn/crashlog/40-watchdog.conf; \
 	fi
+	@if [ -e "$(DESTDIR)/usr/share/acrn/crashlog/80-coredump.conf" ];then \
```
Is it better to check whether there is something residual under /etc/sysctl.d and /etc/systemd/system.conf.d/?

miguelinux replied:
Yes, but lets the crashlogctl do the work.
I'll work on that.

chengangc commented on tools/acrn-crashlog/data/usercrashctl:
```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
```
Can we rename this file as crashlogctl? Since usercrash is part of features of the crashlog.
Thanks.

miguelinux replied:
Sure, will do

chengangc reviewed (commented):
Hi @miguelinux ,

Thanks a lot for your help. Please take a look on my comments. 

Thanks.

miguelinux commented:
Hi @chengangc 

I did the modification, let me know your opinion.

miguelinux commented on tools/acrn-crashlog/Makefile:
```diff
@@ -102,24 +102,27 @@ install:
-		$(RM) $(DESTDIR)/usr/bin/acrnprobe_prepare.sh; \
+	@if [ -e "$(DESTDIR)/usr/bin/crashlogctl" ];then \
+		$(DESTDIR)/usr/bin/crashlogctl disable && \
+		$(RM) $(DESTDIR)/usr/bin/crashlogctl; \
```
Here I call `crashlogctl diable`to erase any installation made by user.
